clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1750,8 +1750,8 @@ private void subscribeInternal(Pattern pattern, Optional<ConsumerRebalanceListen
             throwIfNoAssignorsConfigured();
             log.info("Subscribed to pattern: '{}'", pattern);
             subscriptions.subscribe(pattern, listener);
-            updatePatternSubscription(metadata.fetch());
             metadata.requestUpdateForNewTopics();
+            updatePatternSubscription(metadata.fetch());
```

Comment:
For my understanding, why is the re-ordering necessary?

Reply:
my take, it's not necessary, but it's not harmful either. I wouldn't mind if we prefer to put it back as it it was initially but I expect it truly does not matter.

Reply:
@kirktrue It actually not necessary, it just happened that in the one of the code preview sections of this PR, the call `metadata.requestUpdateForNewTopics` appeared right after `subscriptions.subscribe(pattern, listener)` so I thought the call was previously there, but I also find that it seems more logical to request for metadata update right after we establish the new pattern in `subscriptions`

Reply:
OK. It seems fine either way, so I'm OK to leave it as you have it. Thanks!